docs/_config.yml

```diff
@@ -66,7 +66,7 @@ sphinx:
     # TODO: #917 this path will be the default in sphinx v4
     # mathjax_path: https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js
     # However, it is incompatible with the mathjax config below for macros
-    mathjax_config:
+    mathjax3_config:
```

Comment:
does the configuration key/vals change much? If not, could we leave the key named the same, and then do a translation into the appropriate structure for mathjax 2 or 3 at build time? That way we don't need to break other people's builds.

I guess if we think this will often happen where mathjax breaks its old configs, it'd make sense to have `mathjaxN_config`, but I think it's a better UX if people only have to remember one key name, and we handle the translation to the right config under the hood.

Reply:
thanks @choldgraf this is still in `draft` as I wanted to get the branch up to passing tests and compatible with `sphinx3` and `sphinx4`. This comment is nicely timed as now I am turning attention to the additions we should make from a `jupyter-book` and `usability` perspective. 

> If not, could we leave the key named the same, and then do a translation into the appropriate structure for mathjax 2 or 3 at build time?

We would have to infer `mathjax v2` or `mathjax v3` but this should be achievable. We might just have to parse the `mathjax` cdn that is used in the `sphinx` environment. From `sphinx4` the default will be `mathjax3` but users are able to choose which `mathjax` they use by setting `mathjax_path`. 

My main comment `against` this idea is that this key falls in the `sphinx` configuration section of `_config.yml` and to save confusion I think it would be best to map those values back to `sphinx`. I would think perhaps the best way forward is to add some infer logic to enable builds (checking the condition that if `mathjax 3` is used then ensure `mathjax_config` is swapped to `mathjax3_config`) but issue `warnings` to migrate the `_config.yml` when keys are not set.

Reply:
@choldgraf do you think we need to manage a version agnostic `mathjax` configuration in the jupyter-book config?

https://github.com/executablebooks/jupyter-book/blob/master/jupyter_book/default_config.yml

Reply:
Another quick thought: could we use a single `mathjax:` config key, but use a key like `mathjax_version:` that users provide, that we can use to decide what's the right Sphinx config to use under the hood?

I don't think it'd be a huge deal if users had to manually specify `mathjax2_config` and `mathjax3_config`, but it feels a bit more intuitive to use `mathjax_version` and `mathjax_config` 🤔